src/compiler/crystal/codegen/codegen.cr

```diff
@@ -476,10 +476,23 @@ module Crystal
     def visit(node : TupleLiteral)
       request_value do
         type = node.type.as(TupleInstanceType)
-        @last = allocate_tuple(type) do |tuple_type, i|
-          exp = node.elements[i]
+
+        exp_values = Array({Type, LLVM::Value}).new(type.tuple_types.size)
```

Comment:
I'm a bit worried if this array allocation won't impact in the codegen too much. It might be better to create an array only if there is a splat. Otherwise the existing code works fine with less allocation.